polygeist/include/mlir/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.h

```diff
@@ -389,23 +387,17 @@ inline raw_ostream &operator<<(raw_ostream &os, const OffsetVector &vector) {
 ///   |      | * | | + |  |
 ///   | 0  c3|   |j|   |c4|
 ///
-template <typename OpTy> class MemoryAccess {
+class MemoryAccess {
   template <typename T>
-  friend raw_ostream &operator<<(raw_ostream &, const MemoryAccess<T> &);
+  friend raw_ostream &operator<<(raw_ostream &, const MemoryAccess &);
 
 public:
-  MemoryAccess() = delete;
-
-  template <
-      typename T = OpTy,
-      typename = std::enable_if_t<llvm::is_one_of<T, affine::AffineLoadOp,
-                                                  affine::AffineStoreOp>::value,
-                                  bool>>
-  MemoryAccess(T accessOp, MemoryAccessMatrix &&matrix, OffsetVector &&offsets)
-      : accessOp(accessOp), matrix(std::move(matrix)),
-        offsets(std::move(offsets)) {}
-
-  OpTy getAccessOp() const { return accessOp; }
+  MemoryAccess() = default;
```

Comment:
When do we want to create MemoryAccess without the access matrix and offset vector?

Reply:
The default constructor is required in order to create the DenseMap at line 485 (DenseMap<Operation *, MemoryAccess> accessMap).